.github/workflows/preset-image-build.yml

```diff
@@ -81,31 +86,36 @@ jobs:
         model: ${{fromJson(needs.determine-models.outputs.matrix)}}
       max-parallel: 3
     steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@63c24ba6bd7ba022e95695ff85de572c04a18142 # v2.7.0
+        with:
+          egress-policy: audit
+
       - name: Checkout
-        uses: actions/checkout@v4
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
         with:
           submodules: true
           fetch-depth: 0
-      
+
       - name: Check Available Disk Space
         run: df -h
 
       - name: Install Azure CLI latest
         run: |
-              if ! which az > /dev/null; then
-                  echo "Azure CLI not found. Installing..."
-                  curl -sL https://aka.ms/InstallAzureCLIDeb | sudo bash
-              else
-                  echo "Azure CLI already installed."
-              fi
-    
+          if ! which az > /dev/null; then
+              echo "Azure CLI not found. Installing..."
+              curl -sL https://aka.ms/InstallAzureCLIDeb | sudo bash
+          else
+              echo "Azure CLI already installed."
+          fi
+
       - name: 'Az CLI login'
-        uses: azure/login@v1.6.1
+        uses: azure/login@cb79c773a3cfa27f31f25eb3f677781210c9ce3d # v1.6.1
         with:
-            client-id: ${{ secrets.AZURE_CLIENT_ID }}
-            tenant-id: ${{ secrets.AZURE_TENANT_ID }}
-            allow-no-subscriptions: true
-        
+          client-id: ${{ secrets.AZURE_CLIENT_ID }}
```

Comment:
just small side note: if and when this will get converted into 1es ruler the client and tenanted will not be needed - Depends on the case, I will treat this as outside scope of this PR as this is purely security PR. thanks .